aptos-move/aptos-vm/src/transaction_metadata.rs

```diff
@@ -100,7 +118,7 @@ impl TransactionMetadata {
     }
 
     pub fn is_multi_agent(&self) -> bool {
-        !self.secondary_signers.is_empty()
+        !(self.secondary_signers.is_empty() && self.fee_payer.is_none())
```

Comment:
This seems weird. Why is this necessary when fee payer is a separate tx?

Reply:
@davidiw what do you think is best here

Reply:
This isn't actually checking if it's `multi-agent`, but if it's `multi-signer` or `multi-agent`.  Can you verify where it's using this function?

Reply:
I assume this is needed to know which prologue Move function to call since multiagent txs can now have separate gas payer as well, which means different prologue functions need to call for "old" multiagent txs (no gas payer) vs new. Is this necessary though as the vm logic can just look at the presence of gas payer + whether something is a multiagent tx to decide which function to call?